File: api_demo/spot/Encryption.py
import hashlib
import hmac
import json
import requests
import urllib3
import time
from urllib.parse import quote_plus

def create_order(apiKey,secretKey,symbol,side,order_type,qty,price):
    timestamp = int(time.time() * 10 ** 3)
    url = 'https://api-testnet.bybit.com/spot/v1/order'
    headers = {}
    method="POST"
    params = {
        "side": side,
        "symbol": symbol,
        "type": order_type,
        "qty": qty,
        "price": price,
        "time_in_force": "GoodTillCancel",
        "api_key": apiKey,
        "timestamp": str(timestamp),
        "recv_window": "5000"
    }
    param_str = ''
    for key in sorted(params.keys()):
        v = params[key]
        if isinstance(params[key], bool):
            if params[key]:
                v = 'true'
            else :
                v = 'false'
        param_str += key + '=' + v + '&'
    param_str = param_str[:-1]
    hash = hmac.new(secretKey, param_str.encode("utf-8"), hashlib.sha256)
    signature = hash.hexdigest()
    sign_real = {
        "sign": signature
    }
    param_str = quote_plus(param_str, safe="=&")
    full_param_str = f"{param_str}&sign={sign_real['sign']}"
    urllib3.disable_warnings()
    s = requests.session()
    s.keep_alive = False
    response = requests.request(method, f"{url}?{full_param_str}",headers=headers, verify=False)
    print(response.text)
def main():
    apiKey = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    secret = b"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    create_order(apiKey, secret,'BITUSDT','Buy','Limit','10','1')
    create_order(apiKey, secret,'BITUSDT','Buy','Market','10','0') #for market order, we are just passing the price as 0
if __name__ == '__main__':
    main()

File: api_demo/spot/Encryption.php
<?php

function get_signed_params($public_key, $secret_key, $params) {
    $params = array_merge(['api_key' => $public_key], $params);
    ksort($params);
    //decode return value of http_build_query to make sure signing by plain parameter string
    $signature = hash_hmac('sha256', urldecode(http_build_query($params)), $secret_key);
    return http_build_query($params) . "&sign=$signature";
}

$params = [
	'symbol' => 'BITUSDT', 
	'side' => 'Buy', 
	'type' => 'Limit', 
	'qty' => '10', 
	'price' => '1', 
	'time_in_force' => 'GoodTillCancel',
	'reduce_only' => false,
	'close_on_trigger' => false,
	'timestamp' => time() * 1000
];

$url = 'https://api-testnet.bybit.com/spot/v1/order';

$public_key = 'XXXXXXXXXXXXXXXXXXXXXXX';
$secret_key = 'XXXXXXXXXXXXXXXXXXXXXXX';
$qs=get_signed_params($public_key, $secret_key, $params);
$curl_url=$url."?".$qs;
$curl=curl_init($curl_url);
echo $curl_url . "\n";
curl_setopt($curl, CURLOPT_URL, $curl_url);
#curl_setopt($curl, CURLOPT_POSTFIELDS, $qs);
curl_setopt($curl, CURLOPT_POST, true);
curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($curl, CURLOPT_FOLLOWLOCATION, 1);
curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, 0);
curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, 0);
#curl_setopt($curl, CURLOPT_PROXY,"127.0.0.1:1087");
$response=curl_exec($curl);
echo $response . "\n";
